tools/harness-automation/autothreadharness/harness_case.py

```diff
@@ -733,101 +756,101 @@ def _test_bed(self):
                     ),
                     20,
                 ):
-                    raise Exception('Failed to load TestExecution page')
+                    raise Exception("Failed to load TestExecution page")
             except FailError:
                 raise
             except BaseException:
-                logger.exception('Unexpected error')
+                logger.exception("Unexpected error")
             else:
                 break
 
     def _select_case(self, role, case):
         """Select the test case.
         """
         # select the case
-        elem = Select(self._browser.find_element_by_id('select-dut'))
+        elem = Select(self._browser.find_element_by_id("select-dut"))
         elem.select_by_value(str(role))
         time.sleep(1)
 
         checkbox = None
         wait_until(
             lambda: self._browser.find_elements_by_css_selector(
-                '.tree-node .tree-title'
+                ".tree-node .tree-title"
             )
             and True
         )
         elems = self._browser.find_elements_by_css_selector(
-            '.tree-node .tree-title'
+            ".tree-node .tree-title"
         )
-        finder = re.compile(r'.*\b' + case + r'\b')
-        finder_dotted = re.compile(r'.*\b' + case.replace(' ', r'\.') + r'\b')
+        finder = re.compile(r".*\b" + case + r"\b")
+        finder_dotted = re.compile(r".*\b" + case.replace(" ", r"\.") + r"\b")
         for elem in elems:
             action_chains = ActionChains(self._browser)
             action_chains.move_to_element(elem)
             action_chains.perform()
             logger.debug(elem.text)
             if finder.match(elem.text) or finder_dotted.match(elem.text):
-                parent = elem.find_element_by_xpath('..')
-                checkbox = parent.find_element_by_class_name('tree-checkbox')
+                parent = elem.find_element_by_xpath("..")
+                checkbox = parent.find_element_by_class_name("tree-checkbox")
                 break
 
         if not checkbox:
             time.sleep(5)
-            raise Exception('Failed to find the case')
+            raise Exception("Failed to find the case")
 
         self._browser.execute_script("$('.overview').css('left', '0')")
         checkbox.click()
         time.sleep(1)
 
-        elem = self._browser.find_element_by_id('runTest')
+        elem = self._browser.find_element_by_id("runTest")
         elem.click()
         if not wait_until(
-            lambda: self._browser.find_element_by_id('stopTest') and True, 10
+            lambda: self._browser.find_element_by_id("stopTest") and True, 10
         ):
-            raise Exception('Failed to start test case')
+            raise Exception("Failed to start test case")
 
     def _collect_result(self):
         """Collect test result.
 
         Generate PDF, excel and pcap file
         """
         # generate pdf
-        self._browser.find_element_by_class_name('save-pdf').click()
```

Comment:
Do we still need 'save-pdf' since R1.4.0?

Reply:
if version > R.1.4.0, will not save pdf (line 1153 )

Reply:
For version > = R1.4.0 , will use the self-saved files and not save-as pdf ,excel and pcapng files